File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        Stack<Character> st = new Stack<>();
        int count = 0;

        for(char c : s.toCharArray()) {
            if(c == '[') st.add(c);
            else {
                if(!st.isEmpty()) st.pop();
                else count++;
            }
        }

        return (count + 1)/2;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/2095-minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution(object):
    def minSwaps(self, s):
        st, count = [], 0

        for c in s:
            if c == '[':
                st.append(c)
            else:
                if st:
                    st.pop() 
                else:
                    count += 1
        
        return (count + 1) // 2
        